src/corefx/System.Globalization.Native/pal_collation.c

```diff
@@ -188,15 +188,14 @@ static UCharList* GetCustomRules(int32_t options, UColAttributeValue strength, i
     // Use 512 as the starting size, so the customRules won't have to grow if we are just
     // doing the KanaType custom rule.
     customRules->capacity = 512;
-    customRules->items = calloc(customRules->capacity, sizeof(UChar));
+    customRules->size = 0;
+    customRules->items = malloc(customRules->capacity * sizeof(UChar));
```

Comment:
It is tempting to allocate the actual maximum size of the array here since it's small enough and it gets freed soon after. Thoughts?

Reply:
The comment above it is slightly incorrect. If all loop were executed in the function and all `AddItem` lines got unconditionally called then the upper bound is 88 * 4 (Hiragana rules) and 212 * 5 (Ignore Width rules) elements, for the total of 1412 elements.

Reply:
It would looks like this: https://github.com/filipnavara/coreclr/commit/79b6f032ec5e0476d1b0650c6ba1845b5616043c